// vite-project/src/components/portaladm/modules/servicos/Servicos.jsx
/* eslint-disable no-unused-vars */
import React, { useState, useEffect } from 'react';
import './Servicos.css';
import RelatoriosServicos from './relatorios_servicos';

const Servicos = () => {
  const [showCadastro, setShowCadastro] = useState(false);
  const [showViewServico, setShowViewServico] = useState(false);
  const [showEditServico, setShowEditServico] = useState(false);
  const [showRelatorio, setShowRelatorio] = useState(false);
  const [servicoSelecionado, setServicoSelecionado] = useState(null);
  const [clientes, setClientes] = useState([]);
  const [servicos, setServicos] = useState([]);
  const [servicosFiltrados, setServicosFiltrados] = useState([]);
  const [estatisticas, setEstatisticas] = useState({
    executados: 0,
    agendados: 0,
    vendidos: 0,
    vencidos: 0,
    cancelados: 0,
    emExecucao: 0,
    naoAgendados: 0,
    total: 0
  });
  const [filtros, setFiltros] = useState({
    pesquisa: '',
    status: ''
  });
  const [novoServico, setNovoServico] = useState({
    cliente: '',
    numeroOrcamento: '',
    tipoServico: '',
    descricao: '',
    status: 'Vendido',
    horario: '',
    localEndereco: '',
    cidade: '',
    supervisorResponsavel: '',
    equipeMembros: '',
    veiculoPlaca: '',
    proximaExecucao: '',
    datasExecucao: [],
    estruturasRealizadas: '',
    acompanhanteAssinatura: '',
    observacoes: ''
  });
  const [editFormData, setEditFormData] = useState({});

  const gerarNumeroOrcamento = () => {
    const max = 90000000;
    const min = 10000000;
    if (typeof crypto !== 'undefined' && crypto.getRandomValues) {
      const arr = new Uint32Array(1);
      crypto.getRandomValues(arr);
      return String((arr[0] % max) + min);
    }
    return String(Math.floor(min + Math.random() * max));
  };

  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setNovoServico(prev => ({
      ...prev,
      [name]: value
    }));
  };

  const handleClienteChange = (e) => {
    const { value } = e.target;
    setNovoServico(prev => ({
      ...prev,
      cliente: value
    }));
  };

  const handleAbrirCadastro = () => {
    setShowCadastro(true);
    setNovoServico({
      cliente: '',
      numeroOrcamento: gerarNumeroOrcamento(),
      tipoServico: '',
      descricao: '',
      status: 'Vendido',
      horario: '',
      localEndereco: '',
      cidade: '',
      supervisorResponsavel: '',
      equipeMembros: '',
      veiculoPlaca: '',
      proximaExecucao: '',
      datasExecucao: [],
      estruturasRealizadas: '',
      acompanhanteAssinatura: '',
      observacoes: ''
    });
  };

  // Funções para edição
  const handleRemoverServico = async (servico) => {
    if (window.confirm(`Tem certeza que deseja remover o serviço "${servico.cliente}" - ${servico.tipoServico}?`)) {
      try {
        console.log('🗑️ ID do serviço para deletar:', servico._id);
        console.log('📋 Tipo do ID:', typeof servico._id);
        console.log('📋 Comprimento do ID:', servico._id?.length);
        
        // Tentar DELETE direto no banco via API
        const response = await fetch(`http://localhost:3001/api/Servicos/${servico._id}`, {
          method: 'DELETE',
          headers: {
            'Content-Type': 'application/json',
            'Accept': 'application/json'
          }
        });
        
        console.log('📡 Status da resposta DELETE:', response.status);
        console.log('📡 Status text:', response.statusText);
        
        if (response.ok) {
          const data = await response.json();
          console.log('📄 Resposta sucesso:', data);
          
          if (data && data.success) {
            alert('✅ Serviço removido com sucesso do banco de dados!');
            buscarServicos(); // Atualizar lista
          } else {
            alert(`❌ Erro ao remover: ${data.message || 'Tente novamente'}`);
          }
        } else {
          const errorText = await response.text();
          console.log('❌ Corpo do erro:', errorText);
          
          if (response.status === 404) {
            alert(`❌ Serviço não encontrado no banco (404)\n\nID: ${servico._id}\n\nVerifique se o serviço existe na collection 'servicos'`);
          } else if (response.status === 400) {
            alert(`❌ Requisição inválida (400)\n\n${errorText}\n\nO backend pode não ter DELETE implementado corretamente`);
          } else if (response.status === 405) {
            alert(`❌ Método DELETE não permitido (405)\n\nO backend não suporta DELETE\n\nImplemente: DELETE /api/Servicos/:id`);
          } else {
            alert(`❌ Erro ao remover do banco: ${response.status} ${response.statusText}\n\n${errorText}`);
          }
        }
        
      } catch (error) {
        console.error('❌ Erro na requisição DELETE:', error);
        alert(`❌ Erro de conexão com o banco\n\n${error.message}\n\nVerifique se o backend está conectado ao MongoDB`);
      }
    }
  };

  const handleEditarServico = (servico) => {
    console.log('Abrindo edição para o serviço:', servico);
    console.log('ID do serviço:', servico._id);
    console.log('Datas de execução:', servico.datasExecucao);
    console.log('Estrutura das datas:', JSON.stringify(servico.datasExecucao, null, 2));
    
    // Converter datas ISO para dd/mm/yyyy antes de carregar no formulário
    const converterDataParaBR = (dataStr) => {
      if (!dataStr || dataStr === '') return '';
      
      // Se já estiver no formato dd/mm/yyyy, mantém
      if (dataStr.includes('/')) {
        return dataStr;
      }
      
      // Converter de ISO ou yyyy-mm-dd para dd/mm/yyyy
      if (dataStr.includes('T')) {
        // Formato ISO: 2024-01-15T00:00:00.000Z
        const data = new Date(dataStr);
        return data.toLocaleDateString('pt-BR');
      } else if (dataStr.includes('-')) {
        // Formato yyyy-mm-dd
        const [ano, mes, dia] = dataStr.split('-');
        return `${dia}/${mes}/${ano}`;
      }
      
      return dataStr;
    };
    
    // Preparar dados com datas convertidas
    const servicoFormatado = {
      ...servico,
      proximaExecucao: converterDataParaBR(servico.proximaExecucao),
      datasExecucao: (servico.datasExecucao || []).map(data => ({
        ...data,
        data: converterDataParaBR(data.data)
      }))
    };
    
    console.log('Serviço formatado para edição:', servicoFormatado);
    
    setServicoSelecionado(servico);
    setEditFormData(servicoFormatado);
    setShowEditServico(true);
    // Prevenir scroll do body
    document.body.style.overflow = 'hidden';
  };

  const handleEditarChange = (e) => {
    const { name, value } = e.target;
    setEditFormData(prev => ({
      ...prev,
      [name]: value
    }));
  };

  const handleSalvarEdicao = async (e) => {
    e.preventDefault();
    
    try {
      // 1. Verificar se o backend está online
      console.log('🔍 Verificando se o backend está online...');
      const healthCheck = await fetch('http://localhost:3001/api/Servicos');
      
      if (!healthCheck.ok) {
        throw new Error('Backend não está respondendo corretamente');
      }
      
      console.log('✅ Backend está online!');
      console.log('📋 ID do serviço:', servicoSelecionado._id);
      
      // Converter datas para formato ISO antes de enviar para o backend
      const converterDataParaISO = (dataStr) => {
        if (!dataStr || dataStr === '') return null;
        
        // Se já estiver no formato ISO, retorna como está
        if (dataStr.includes('T')) {
          return dataStr;
        }
        
        // Converter de dd/mm/yyyy para yyyy-mm-ddT00:00:00.000Z
        if (dataStr.includes('/')) {
          const [dia, mes, ano] = dataStr.split('/');
          const data = new Date(ano, mes - 1, dia);
          return data.toISOString();
        }
        
        // Se for yyyy-mm-dd, converter para ISO
        if (dataStr.includes('-')) {
          const [ano, mes, dia] = dataStr.split('-');
          const data = new Date(ano, mes - 1, dia);
          return data.toISOString();
        }
        
        return dataStr;
      };

      // Preparar dados com datas convertidas
      const dadosParaEnviar = {
        ...editFormData,
        proximaExecucao: converterDataParaISO(editFormData.proximaExecucao),
        datasExecucao: (editFormData.datasExecucao || []).map(data => ({
          ...data,
          data: converterDataParaISO(data.data)
        }))
      };
      
      console.log('📊 Dados a serem salvos (com datas convertidas):', dadosParaEnviar);
      
      // 2. Tentar diferentes métodos de atualização
      const apiUrl = `http://localhost:3001/api/Servicos/${servicoSelecionado._id}`;
      console.log('🌐 Tentando atualizar em:', apiUrl);
      
      // Método 1: PUT (padrão REST)
      try {
        console.log('🔄 Tentando método PUT...');
        const putResponse = await fetch(apiUrl, {
          method: 'PUT',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(dadosParaEnviar)
        });
        
        if (putResponse.ok) {
          const data = await putResponse.json();
          if (data && data.success) {
            alert('✅ Serviço atualizado com sucesso!');
            setShowEditServico(false);
            document.body.style.overflow = 'unset';
            buscarServicos();
            return;
          }
        }
        console.log('❌ PUT não funcionou, status:', putResponse.status);
      } catch (putError) {
        console.log('❌ Erro no PUT:', putError.message);
      }
      
      // Método 2: PATCH (atualização parcial)
      try {
        console.log('🔄 Tentando método PATCH...');
        const patchResponse = await fetch(apiUrl, {
          method: 'PATCH',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify(dadosParaEnviar)
        });
        
        if (patchResponse.ok) {
          const data = await patchResponse.json();
          if (data && data.success) {
            alert('✅ Serviço atualizado com sucesso (PATCH)!');
            setShowEditServico(false);
            document.body.style.overflow = 'unset';
            buscarServicos();
            return;
          }
        }
        console.log('❌ PATCH não funcionou, status:', patchResponse.status);
      } catch (patchError) {
        console.log('❌ Erro no PATCH:', patchError.message);
      }
      
      // Método 3: POST com ID no corpo (alternativa)
      try {
        console.log('🔄 Tentando método POST com ID...');
        const postResponse = await fetch('http://localhost:3001/api/Servicos', {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            ...dadosParaEnviar,
            _id: servicoSelecionado._id, // Incluir ID para identificação
            operacao: 'atualizar' // Flag para o backend saber que é atualização
          })
        });
        
        if (postResponse.ok) {
          const data = await postResponse.json();
          if (data && data.success) {
            alert('✅ Serviço atualizado com sucesso (POST)!');
            setShowEditServico(false);
            document.body.style.overflow = 'unset';
            buscarServicos();
            return;
          }
        }
        console.log('❌ POST não funcionou, status:', postResponse.status);
      } catch (postError) {
        console.log('❌ Erro no POST:', postError.message);
      }
      
      // Se nenhum método funcionou
      throw new Error('Nenhum método de atualização funcionou. O backend precisa implementar PUT, PATCH ou POST com ID para atualização.');
      
    } catch (error) {
      console.error('❌ Erro completo:', error);
      
      if (error.message.includes('Failed to fetch')) {
        alert('❌ Backend não está rodando!\n\nSoluções:\n1. Inicie o backend: npm run dev\n2. Verifique se está na porta 3001\n3. Teste: http://localhost:3001/api/Servicos\n\nO frontend não consegue se conectar ao servidor.');
      } else {
        alert(`❌ Erro: ${error.message}\n\nPara atualizar serviços no banco "Servicos" collection "servico", o backend precisa implementar:\n- PUT /api/Servicos/:id\n- Ou PATCH /api/Servicos/:id\n- Ou POST /api/Servicos com ID no corpo`);
      }
    }
  };

  const handleCancelarEdicao = () => {
    setShowEditServico(false);
    setEditFormData({});
    // Restaurar scroll do body
    document.body.style.overflow = 'unset';
  };

  const handleAdicionarDataExecucao = () => {
    setNovoServico(prev => ({
      ...prev,
      datasExecucao: [...prev.datasExecucao, '']
    }));
  };

  const handleRemoverDataExecucao = (index) => {
    setNovoServico(prev => ({
      ...prev,
      datasExecucao: prev.datasExecucao.filter((_, i) => i !== index)
    }));
  };

  const handleDataExecucaoChange = (index, campo, value) => {
    setNovoServico(prev => {
      const novasDatas = [...prev.datasExecucao];
      if (!novasDatas[index]) {
        novasDatas[index] = {};
      }
      
      novasDatas[index] = {
        ...novasDatas[index],
        [campo]: value
      };
      return { ...prev, datasExecucao: novasDatas };
    });
  };

  const handleCancelar = () => {
    setShowCadastro(false);
    setNovoServico({
      cliente: '',
      numeroOrcamento: '',
      tipoServico: '',
      descricao: '',
      status: 'Vendido',
      horario: '',
      localEndereco: '',
      cidade: '',
      supervisorResponsavel: '',
      equipeMembros: '',
      veiculoPlaca: '',
      proximaExecucao: '',
      datasExecucao: [],
      estruturasRealizadas: '',
      acompanhanteAssinatura: '',
      observacoes: ''
    });
  };

  const handleVerServico = (servico) => {
    setServicoSelecionado(servico);
    setShowViewServico(true);
  };

  const handleCloseView = () => {
    setShowViewServico(false);
    setServicoSelecionado(null);
  };

  const buscarClientes = async () => {
    try {
      console.log('🔍 Iniciando busca de clientes...');
      const response = await fetch('http://localhost:3001/api/clientes');
      console.log('📡 Status HTTP:', response.status);
      console.log('📡 Headers:', response.headers);
      
      if (!response.ok) {
        console.error('❌ Resposta não OK:', response.status, response.statusText);
        return;
      }
      
      const data = await response.json();
      console.log('📋 Dados recebidos:', data);
      console.log('📊 Estrutura completa:', JSON.stringify(data, null, 2));
      
      // Tentar diferentes formatos de resposta
      if (data.success) {
        console.log('✅ Sucesso na API (formato success)');
        console.log('👥 Array clientes:', data.clientes);
        console.log('📊 Quantidade:', data.clientes?.length || 0);
        setClientes(data.clientes || []);
      } else if (Array.isArray(data)) {
        // Se retornar diretamente um array
        console.log('✅ Sucesso na API (formato array)');
        console.log('👥 Array clientes:', data);
        console.log('📊 Quantidade:', data.length);
        setClientes(data);
      } else if (data.clientes && Array.isArray(data.clientes)) {
        // Se tiver clientes mas success for false
        console.log('✅ Clientes encontrados (sem success)');
        console.log('👥 Array clientes:', data.clientes);
        console.log('📊 Quantidade:', data.clientes.length);
        setClientes(data.clientes);
      } else {
        console.error('❌ API retornou erro:', data.message);
        console.error('📄 Resposta completa:', data);
        console.error('🔍 Formato inesperado - verifique estrutura acima');
      }
    } catch (error) {
      console.error('❌ Erro na requisição:', error);
      console.error('🔍 Detalhes:', error.message);
    }
  };

  const buscarServicos = async (filtrosParaBuscar) => {
    try {
      console.log('🔍 Buscando serviços do banco Servicos...');
      let url = 'http://localhost:3001/api/Servicos';
      if (filtrosParaBuscar) {
        const params = new URLSearchParams();
        Object.entries(filtrosParaBuscar).forEach(([key, value]) => {
          if (value !== undefined && value !== null && String(value).trim() !== '') {
            params.set(key, value);
          }
        });
        const queryString = params.toString();
        if (queryString) url = `${url}?${queryString}`;
      }

      const response = await fetch(url);
      const data = await response.json();
      
      if (data.success) {
        console.log(`📋 Encontrados ${data.servicos.length} serviços`);
        setServicos(data.servicos);
      } else {
        console.error('Erro ao buscar serviços:', data.message);
      }
    } catch (error) {
      console.error('Erro ao buscar serviços:', error);
    }
  };

  const calcularEstatisticas = (listaServicos) => {
    console.log(`📊 Calculando estatísticas de ${listaServicos.length} serviços totais`);
    
    const stats = {
      executados: 0,
      agendados: 0,
      vendidos: 0,
      vencidos: 0,
      cancelados: 0,
      emExecucao: 0,
      naoAgendados: 0,
      total: 0
    };

    listaServicos.forEach(servico => {
      const status = servico.status;

      switch (status) {
        case 'Executado':
          stats.executados++;
          break;
        case 'Agendado':
          stats.agendados++;
          break;
        case 'Vendido':
          stats.vendidos++;
          break;
        case 'Nao Agendado':
          stats.naoAgendados++;
          break;
        case 'Em execucao':
          stats.emExecucao++;
          break;
        case 'Cancelado':
          stats.cancelados++;
          break;
        case 'Vencido':
          stats.vencidos++;
          break;
      }
    });

    // Calcular total
    stats.total = listaServicos.length;

    console.log(`📊 Estatísticas totais:`, stats);
    setEstatisticas(stats);
  };

  const handleFiltroChange = (e) => {
    const { name, value } = e.target;
    setFiltros(prev => ({
      ...prev,
      [name]: value
    }));
  };

  useEffect(() => {
    if (showCadastro) {
      buscarClientes();
    }
  }, [showCadastro]);

  // Buscar serviços quando o componente montar
  useEffect(() => {
    buscarServicos();
  }, []);

  // Atualizar serviços filtrados quando os serviços mudarem
  useEffect(() => {
    setServicosFiltrados(servicos);
  }, [servicos]);

  // Calcular estatísticas quando os serviços mudarem
  useEffect(() => {
    calcularEstatisticas(servicos);
  }, [servicos]);

  useEffect(() => {
    const handler = setTimeout(() => {
      buscarServicos({ pesquisa: filtros.pesquisa, status: filtros.status });
    }, 250);
    return () => clearTimeout(handler);
  }, [filtros.pesquisa, filtros.status]);

  const handleLimparFiltros = async () => {
    const filtrosLimpos = {
      pesquisa: '',
      status: ''
    };
    setFiltros(filtrosLimpos);
    await buscarServicos();
  };

  const handleCriarServico = async (e) => {
    e.preventDefault();
    
    try {
      // Validar campos obrigatórios
      
      // Converter datas para formato ISO antes de enviar para o backend
      const converterDataParaISO = (dataStr) => {
        if (!dataStr || dataStr === '') return null;
        
        // Se já estiver no formato ISO, retorna como está
        if (dataStr.includes('T')) {
          return dataStr;
        }
        
        // Converter de dd/mm/yyyy para yyyy-mm-ddT00:00:00.000Z
        if (dataStr.includes('/')) {
          const [dia, mes, ano] = dataStr.split('/');
          const data = new Date(ano, mes - 1, dia);
          return data.toISOString();
        }
        
        // Se for yyyy-mm-dd, converter para ISO
        if (dataStr.includes('-')) {
          const [ano, mes, dia] = dataStr.split('-');
          const data = new Date(ano, mes - 1, dia);
          return data.toISOString();
        }
        
        return dataStr;
      };

      // Preparar dados para enviar ao backend
      const servicoData = {
        cliente: novoServico.cliente,
        numeroOrcamento: novoServico.numeroOrcamento,
        tipoServico: novoServico.tipoServico,
        descricao: novoServico.descricao,
        status: novoServico.status,
        horario: novoServico.horario,
        localEndereco: novoServico.localEndereco,
        cidade: novoServico.cidade,
        supervisorResponsavel: novoServico.supervisorResponsavel,
        equipeMembros: novoServico.equipeMembros,
        veiculoPlaca: novoServico.veiculoPlaca,
        proximaExecucao: converterDataParaISO(novoServico.proximaExecucao),
        datasExecucao: (novoServico.datasExecucao || []).map(data => ({
          ...data,
          data: converterDataParaISO(data.data)
        })),
        estruturasRealizadas: novoServico.estruturasRealizadas,
        acompanhanteAssinatura: novoServico.acompanhanteAssinatura,
        observacoes: novoServico.observacoes
      };

      console.log('💾 Enviando todos os campos para a collection servico:', servicoData);

      // Enviar para API
      const response = await fetch('http://localhost:3001/api/Servicos', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(servicoData)
      });

      const data = await response.json();

      if (response.ok && data.success) {
        alert('✅ Serviço criado com sucesso!');
        handleCancelar();
        buscarServicos(); // Atualizar lista de serviços
      } else {
        alert(`❌ Erro ao criar serviço: ${data.message || 'Tente novamente'}`);
      }
    } catch (error) {
      console.error('Erro ao criar serviço:', error);
      alert('❌ Erro de conexão. Tente novamente.');
    }
  };

  return (
    <div className="portaladm-servicos">
      {/* Cabeçalho - Escondido na visualização */}
      {!showViewServico && (
        <div className="servicos-header">
          <h2>Gestão de Serviços</h2>
          {!showCadastro && (
            <button className="btn-primary" onClick={handleAbrirCadastro}>
              + Novo Serviço
            </button>
          )}
        </div>
      )}

      {/* Formulário de Cadastro na Mesma Tela */}
      {showCadastro && (
        <>
          <div className="cadastro-overlay" />
          <div className="cadastro-servico">
          <div className="cadastro-header">
            <h3>Novo Serviço</h3>
            <button className="btn-fechar-cadastro" onClick={handleCancelar}>
              ✕
            </button>
          </div>
          
          <form onSubmit={handleCriarServico} className="servico-form">
            <div className="form-row">
              <div className="form-group">
                <label htmlFor="cliente">Cliente *</label>
                <select
                  id="cliente"
                  name="cliente"
                  value={novoServico.cliente}
                  onChange={handleClienteChange}
                  required
                >
                  <option value="">Selecione cliente...</option>
                  {clientes.map((cliente) => (
                    <option key={cliente._id || cliente.id} value={cliente.razaoSocial || cliente.nome}>
                      {cliente.razaoSocial || cliente.nome}
                    </option>
                  ))}
                </select>
              </div>
              
              <div className="form-group">
                <label htmlFor="numeroOrcamento">Nº Orçamento</label>
                <input
                  type="text"
                  id="numeroOrcamento"
                  name="numeroOrcamento"
                  value={novoServico.numeroOrcamento}
                  onChange={handleInputChange}
                />
              </div>
            </div>

            <div className="form-row">
              <div className="form-group">
                <label htmlFor="tipoServico">Tipo de Serviço *</label>
                <select
                  id="tipoServico"
                  name="tipoServico"
                  value={novoServico.tipoServico}
                  onChange={handleInputChange}
                  required
                >
                  <option value="">Selecione...</option>
                  <option value="Limpeza de reservatorio">Limpeza de Reservatórios</option>
                  <option value="Desinfeccao de Bebedouros">Desinfecção de Bebedouros</option>
                  <option value="Limpeza de Coifa">Limpeza de Coifa</option>
                  <option value="Limpeza de Silo">Limpeza de Silo</option>
                  <option value="Limpeza de Tanque">Limpeza de Tanques</option>
                  <option value="Sanitizacao">Sanitização</option>
                  <option value="Coleta de Agua">Coleta de Água</option>
                  <option value="Integracao">Integração</option>
                  <option value="Outros">Outros</option>
                </select>
              </div>
              
              <div className="form-group">
                <label htmlFor="status">Status:</label>
                <select
                  id="status"
                  name="status"
                  value={novoServico.status}
                  onChange={handleInputChange}
                >
                  <option value="Vendido">Vendido</option>
                  <option value="Agendado">Agendado</option>
                  <option value="Nao Agendado">Não Agendado</option>
                  <option value="Em execucao">Em execução</option>
                  <option value="Executado">Executado</option>
                  <option value="Vencido">Vencido</option>
                  <option value="Cancelado">Cancelado</option>
                </select>
              </div>
            </div>

            <div className="form-row">
              <div className="form-group">
                <label htmlFor="descricao">Descrição do Serviço</label>
                <textarea
                  id="descricao"
                  name="descricao"
                  value={novoServico.descricao}
                  onChange={handleInputChange}
                  placeholder="Estruturas realizadas, observações..."
                  rows="3"
                />
              </div>
            </div>

            <div className="form-row">
              <div className="form-group">
                <label htmlFor="horario">Horário</label>
                <input
                  type="text"
                  id="horario"
                  name="horario"
                  value={novoServico.horario}
                  onChange={handleInputChange}
                  placeholder="Ex: 08:00"
                />
              </div>
              
              <div className="form-group">
                <label htmlFor="localEndereco">Local / Endereço do Serviço</label>
                <input
                  type="text"
                  id="localEndereco"
                  name="localEndereco"
                  value={novoServico.localEndereco}
                  onChange={handleInputChange}
                />
              </div>
            </div>

            <div className="form-row">
              <div className="form-group">
                <label htmlFor="cidade">Cidade</label>
                <input
                  type="text"
                  id="cidade"
                  name="cidade"
                  value={novoServico.cidade}
                  onChange={handleInputChange}
                />
              </div>
              
              <div className="form-group">
                <label htmlFor="supervisorResponsavel">Supervisor Responsável</label>
                <input
                  type="text"
                  id="supervisorResponsavel"
                  name="supervisorResponsavel"
                  value={novoServico.supervisorResponsavel}
                  onChange={handleInputChange}
                />
              </div>
            </div>

            <div className="form-row">
              <div className="form-group">
                <label htmlFor="equipeMembros">Equipe / Membros</label>
                <input
                  type="text"
                  id="equipeMembros"
                  name="equipeMembros"
                  value={novoServico.equipeMembros}
                  onChange={handleInputChange}
                  placeholder="Nomes separados por vírgula"
                />
              </div>
              
              <div className="form-group">
                <label htmlFor="veiculoPlaca">Veículo (Placa)</label>
                <input
                  type="text"
                  id="veiculoPlaca"
                  name="veiculoPlaca"
                  value={novoServico.veiculoPlaca}
                  onChange={handleInputChange}
                />
              </div>
            </div>

            <div className="form-row">
              <div className="form-group">
                <label htmlFor="proximaExecucao">Próxima Execução</label>
                <input
                  type="text"
                  id="proximaExecucao"
                  name="proximaExecucao"
                  placeholder="dd/mm/aaaa"
                  maxLength="10"
                  value={novoServico.proximaExecucao || ''}
                  onChange={(e) => {
                    let value = e.target.value;
                    
                    // Aplicar máscara dd/mm/yyyy
                    value = value.replace(/\D/g, ''); // Remove não números
                    if (value.length >= 2) {
                      value = value.slice(0, 2) + '/' + value.slice(2);
                    }
                    if (value.length >= 5) {
                      value = value.slice(0, 5) + '/' + value.slice(5);
                    }
                    if (value.length > 10) {
                      value = value.slice(0, 10);
                    }
                    
                    handleInputChange({
                      target: {
                        name: 'proximaExecucao',
                        value: value
                      }
                    });
                  }}
                />
              </div>
              <div className="form-group">
                <label>&nbsp;</label>
                <button
                  type="button"
                  className="btn-adicionar-data"
                  onClick={handleAdicionarDataExecucao}
                >
                  + Adicionar Data de Execução
                </button>
              </div>
            </div>

            {novoServico.datasExecucao.length > 0 && (
            <div className="form-row">
              <div className="form-group">
                <label>Datas de Execução</label>
                <div className="datas-execucao">
                  {novoServico.datasExecucao.map((data, index) => (
                    <div key={index} className="data-execucao-item">
                      <input
                        type="text"
                        placeholder="dd/mm/aaaa"
                        maxLength="10"
                        value={data.data || ''}
                        onChange={(e) => {
                          let value = e.target.value;
                          
                          // Aplicar máscara dd/mm/yyyy
                          value = value.replace(/\D/g, ''); // Remove não números
                          if (value.length >= 2) {
                            value = value.slice(0, 2) + '/' + value.slice(2);
                          }
                          if (value.length >= 5) {
                            value = value.slice(0, 5) + '/' + value.slice(5);
                          }
                          if (value.length > 10) {
                            value = value.slice(0, 10);
                          }
                          
                          handleDataExecucaoChange(index, 'data', value);
                        }}
                      />
                      <select className="status-data" value={data.status || ''} onChange={(e) => handleDataExecucaoChange(index, 'status', e.target.value)}>
                        <option value="">Status</option>
                        <option value="Programado">Programado</option>
                        <option value="Executado">Executado</option>
                        <option value="Cancelado">Cancelado</option>
                      </select>
                      <input
                        type="text"
                        value={data.observacao || ''}
                        onChange={(e) => handleDataExecucaoChange(index, 'observacao', e.target.value)}
                        placeholder="Observação"
                      />
                      <button
                        type="button"
                        className="btn-remover-data"
                        onClick={() => handleRemoverDataExecucao(index)}
                      >
                        <svg width="14" height="14" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round">
  <polyline points="3 6 5 6 21 6"></polyline>
  <path d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"></path>
</svg>
                      </button>
                    </div>
                  ))}
                </div>
              </div>
            </div>
          )}
          
          
            <div className="form-row">
              <div className="form-group">
                <label htmlFor="estruturasRealizadas">Estruturas Realizadas</label>
                <textarea
                  id="estruturasRealizadas"
                  name="estruturasRealizadas"
                  value={novoServico.estruturasRealizadas}
                  onChange={handleInputChange}
                  placeholder="Ex: 1X INF. 25.000L + 1X SUP. 15.000L"
                />
              </div>
            </div>

            <div className="form-row">
              <div className="form-group">
                <label htmlFor="acompanhanteAssinatura">Acompanhante / Assinatura</label>
                <input
                  type="text"
                  id="acompanhanteAssinatura"
                  name="acompanhanteAssinatura"
                  value={novoServico.acompanhanteAssinatura}
                  onChange={handleInputChange}
                />
              </div>
            </div>

            <div className="form-row">
              <div className="form-group">
                <label htmlFor="observacoes">Observações do Serviço</label>
                <textarea
                  id="observacoes"
                  name="observacoes"
                  value={novoServico.observacoes}
                  onChange={handleInputChange}
                  rows="4"
                />
              </div>
            </div>

            <div className="form-actions">
              <button type="button" className="btn-cancelar" onClick={handleCancelar}>
                Cancelar
              </button>
              <button type="submit" className="btn-criar">
                Criar Serviço
              </button>
            </div>
          </form>
        </div>
        </>
      )}
      
      {/* Estatísticas - Escondidas na visualização */}
      {!showViewServico && (
        <div className="servicos-stats">
          <div className="stat-box stat-executados">
            <div className="stat-number">{estatisticas.executados}</div>
            <div className="stat-label">Executados</div>
          </div>
          <div className="stat-box stat-agendados">
            <div className="stat-number">{estatisticas.agendados}</div>
            <div className="stat-label">Agendados</div>
          </div>
          <div className="stat-box stat-vendidos">
            <div className="stat-number">{estatisticas.vendidos}</div>
            <div className="stat-label">Vendidos</div>
          </div>
          <div className="stat-box stat-vencidos">
            <div className="stat-number">{estatisticas.vencidos}</div>
            <div className="stat-label">Vencidos</div>
          </div>
          <div className="stat-box stat-cancelados">
            <div className="stat-number">{estatisticas.cancelados}</div>
            <div className="stat-label">Cancelados</div>
          </div>
          <div className="stat-box stat-em-execucao">
            <div className="stat-number">{estatisticas.emExecucao}</div>
            <div className="stat-label">Em Execução</div>
          </div>
          <div className="stat-box stat-nao-agendados">
            <div className="stat-number">{estatisticas.naoAgendados}</div>
            <div className="stat-label">Não Agendados</div>
          </div>
          <div className="stat-box stat-total">
            <div className="stat-number">{estatisticas.total}</div>
            <div className="stat-label">Total de Serviços</div>
          </div>
        </div>
      )}

      {/* Filtros - Escondidos na visualização */}
      {!showViewServico && (
        <div className="servicos-filters">
          <div className="search-container">
            <svg className="search-icon" width="16" height="16" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
              <path d="M21 21L15.5 15.5M17 10C17 13.866 13.866 17 10 17C6.13401 17 3 13.866 3 10C3 6.13401 6.13401 3 10 3C13.866 3 17 6.13401 17 10Z" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
            </svg>
            <input 
              type="text" 
              className="search-input" 
              name="pesquisa"
              value={filtros.pesquisa}
              onChange={handleFiltroChange}
              placeholder="Pesquisar cliente,serviço,orçamento"
            />
          </div>

          <div className="filter-group">
            <select
              className="filter-select"
              name="status"
              value={filtros.status}
              onChange={handleFiltroChange}
            >
              <option value="">Todos Status</option>
              <option value="Executado">Executado</option>
              <option value="Agendado">Agendado</option>
              <option value="Vendido">Vendido</option>
              <option value="Nao Agendado">Não Agendado</option>
              <option value="Em execucao">Em execução</option>
              <option value="Vencido">Vencido</option>
              <option value="Cancelado">Cancelado</option>
            </select>
          </div>
          <button className="btn-secondary" onClick={handleLimparFiltros}>
            Limpar
          </button>
        </div>
      )}

      {/* Conteúdo Principal: Lista ou Visualização */}
      {showViewServico && servicoSelecionado ? (
        /* Visualização do Serviço */
        <div className="servico-view-page">
          {/* Cabeçalho Superior */}
          <div className="view-header-superior">
            <div style={{ display: 'flex', alignItems: 'center', gap: '1rem' }}>
              <button className="btn-secondary" onClick={handleCloseView}>
                ← Voltar
              </button>
              <div>
                <div className="cliente-destaque">{servicoSelecionado.cliente?.toUpperCase() || 'N/A'}</div>
                <div className="orcamento-pequeno">Orçamento: {servicoSelecionado.numeroOrcamento || 'N/A'}</div>
              </div>
            </div>
            
            <div className="status-servico-quadro">
              <label></label>
              <div className={`status-badge status-${servicoSelecionado.status?.toLowerCase().replace(' ', '-')}`}>
                {servicoSelecionado.status?.toUpperCase() || 'N/A'}
              </div>
            </div>
            
            <div className="tipo-servico-quadro">
              <label>Tipo de Serviço</label>
              <div>{servicoSelecionado.tipoServico || 'N/A'}</div>
            </div>
          </div>

          

          {/* Campos em Linha */}
          <div className="campos-linha">
            <div className="campo-branco">
              <label>Supervisor Responsável</label>
              <div className="campo-valor">{servicoSelecionado.supervisorResponsavel || 'N/A'}</div>
            </div>
            <div className="campo-branco">
              <label>Veículo/Placa</label>
              <div className="campo-valor">{servicoSelecionado.veiculoPlaca || 'N/A'}</div>
            </div>
            <div className="campo-branco">
              <label>Horário</label>
              <div className="campo-valor">{servicoSelecionado.horario || 'N/A'}</div>
            </div>
            <div className="campo-branco">
              <label>Local/Endereço</label>
              <div className="campo-valor">{servicoSelecionado.localEndereco || 'N/A'}</div>
            </div>
          </div>

          {/* Datas, Status e Observações - Tudo no mesmo quadro */}
          <div className="datas-status-obs">
            <div className="todos-juntos">
              <div className="datas-execucao">
                <label>Datas de Execução</label>
                <div className="datas-list">
                  {servicoSelecionado.datasExecucao && servicoSelecionado.datasExecucao.length > 0 
                    ? servicoSelecionado.datasExecucao.map((data, index) => (
                        <div key={index} className="execution-item-view">
                          <span className="date-tag-view">
                            {new Date(data.data).toLocaleDateString('pt-BR')}
                          </span>

                          {data.observacao && (
                            <div className="observacao-text-view">
                              {data.observacao}
                            </div>
                          )}
                        </div>
                      ))
                    : 'Nenhuma data registrada'
                  }
                </div>
              </div>
              
              <div className="status-info">
                <label>Status da Data</label>
                <div className={`status-badge status-${(servicoSelecionado.datasExecucao?.[0]?.status || 'N/A').toLowerCase().replace(' ', '-')}`}>
                  {(servicoSelecionado.datasExecucao?.[0]?.status || 'N/A').toUpperCase()}
                </div>
              </div>
              
              <div className="observacoes-info">
                <label>Observações</label>
                <div className="obs-texto">{servicoSelecionado.observacoes || 'Nenhuma observação informada'}</div>
              </div>
            </div>
          </div>

          {/* Demais Campos */}
          <div className="demais-campos">
            <div className="campo-grupo">
              <label>Valor do Serviço</label>
              <div className="campo-valor-destaque">R$ {servicoSelecionado.valor || '0,00'}</div>
            </div>
            
            
            
            <div className="campo-grupo">
              <label>Periodicidade</label>
              <div className="campo-valor">{servicoSelecionado.periodicidade || 'N/A'}</div>
            </div>
            
            <div className="campo-grupo">
              <label>Equipe Membros</label>
              <div className="campo-valor">{servicoSelecionado.equipeMembros || 'N/A'}</div>
            </div>
            
            <div className="campo-grupo">
              <label>Próxima Execução</label>
              <div className="campo-valor">
                {servicoSelecionado.proximaExecucao ? 
                  // Se já estiver no formato dd/mm/yyyy, mantém; senão converte
                  servicoSelecionado.proximaExecucao.includes('/') ?
                    servicoSelecionado.proximaExecucao :
                    new Date(servicoSelecionado.proximaExecucao).toLocaleDateString('pt-BR')
                  : 'N/A'
                }
              </div>
            </div>
            
            <div className="campo-grupo">
              <label>Estruturas Realizadas</label>
              <div className="campo-valor">{servicoSelecionado.estruturasRealizadas || 'N/A'}</div>
            </div>
            
            <div className="campo-grupo">
              <label>Acompanhante / Assinatura</label>
              <div className="campo-valor">{servicoSelecionado.acompanhanteAssinatura || 'N/A'}</div>
            </div>
            
            <div className="campo-grupo">
              <label>Descrição do Serviço</label>
              <div className="descricao-texto">{servicoSelecionado.descricao || 'Nenhuma descrição informada'}</div>
            </div>
            
            <div className="campo-grupo">
              <label>Data de Cadastro</label>
              <div className="campo-valor">{servicoSelecionado.createdAt ? new Date(servicoSelecionado.createdAt).toLocaleDateString('pt-BR') : 'N/A'}</div>
            </div>
            
            <div className="campo-grupo">
              <label>Última Atualização</label>
              <div className="campo-valor">{servicoSelecionado.updatedAt ? new Date(servicoSelecionado.updatedAt).toLocaleDateString('pt-BR') : 'N/A'}</div>
            </div>
            
            <button 
              className="btn-relatorio"
              onClick={() => setShowRelatorio(true)}
              title="Gerar Relatório de Serviços"
            >
              <i className="fas fa-file-lines"></i>
              Gerar Relatório
            </button>
          </div>
        </div>
      ) : (
        /* Lista de Serviços */
        <div className="servicos-table-container">
          <table className="servicos-table">
            <thead>
              <tr>
                <th style={{ textAlign: 'center' }}>Código</th>
                <th style={{ textAlign: 'center' }}>Cliente</th>
                <th style={{ textAlign: 'center' }}>Tipo</th>
                <th style={{ textAlign: 'center' }}>Data Execução</th>
                <th style={{ textAlign: 'center' }}>Próxima</th>
                <th style={{ textAlign: 'center' }}>Status</th>
                <th style={{ textAlign: 'center' }}>Ações</th>
              </tr>
            </thead>
            <tbody>
              {servicosFiltrados.map((servico, index) => (
                <tr 
                  key={servico._id}
                  style={{
                    backgroundColor: index % 2 === 0 ? '#ac4e4e1e' : '#f7f7f7'
                  }}
                >
                  <td style={{ textAlign: 'center' }}>SRV-{String(index + 1).padStart(3, '0')}</td>
                  <td style={{ textAlign: 'center' }}>{servico.cliente.toUpperCase()}</td>
                  <td style={{ textAlign: 'center' }}>{servico.tipoServico}</td>
                  <td style={{ textAlign: 'center' }}>
                    {servico.datasExecucao && servico.datasExecucao.length > 0 
                      ? servico.datasExecucao.map((data, dataIndex) => {
                          try {
                            let dataValor = data?.data || data || '';
                            
                            if (!dataValor) {
                              return null;
                            }
                            
                            if (typeof dataValor !== 'string') {
                              return null;
                            }
                            
                            let dataFormatada = '';
                            
                            if (dataValor.includes('T')) {
                              // Formato ISO
                              const dateObj = new Date(dataValor);
                              if (isNaN(dateObj.getTime())) {
                                return null;
                              }
                              dataFormatada = dateObj.toLocaleDateString('pt-BR');
                            } else if (dataValor.includes('-')) {
                              // Formato yyyy-mm-dd
                              const [ano, mes, dia] = dataValor.split('-');
                              if (ano && mes && dia && ano.length === 4 && mes.length === 2 && dia.length === 2) {
                                dataFormatada = `${dia}/${mes}/${ano}`;
                              } else {
                                return null;
                              }
                            } else if (dataValor.includes('/')) {
                              // Já está em formato brasileiro
                              const partes = dataValor.split('/');
                              if (partes.length === 3) {
                                dataFormatada = dataValor;
                              } else {
                                return null;
                              }
                            } else {
                              return null;
                            }
                            
                            return dataFormatada;
                          } catch (error) {
                            return null;
                          }
                        }).filter(d => d !== null && d !== undefined && d !== '').join(', ') || 'N/A'
                      : 'N/A'
                    }
                  </td>
                  <td style={{ textAlign: 'center' }}>{servico.proximaExecucao ? new Date(servico.proximaExecucao).toLocaleDateString('pt-BR') : 'N/A'}</td>
                  <td 
                    className={`status-${servico.status.toLowerCase().replace(' ', '-')}`}
                    style={{
                      color: '#7a7e7afa',
                      fontWeight: 'bold',
                      backgroundColor: '#c2c4c24d' // Verde igual ao Agendado
                    }}
                  >
                    {servico.status.toUpperCase()}
                  </td>
                  <td style={{ textAlign: 'center' }}>
                    <button className="btn-action" title="Ver" onClick={() => handleVerServico(servico)}>
                      <svg width="16" height="16" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                        <path d="M1 12s4-8 11-8 11 8 11 8-4 8-11 8-11-8-11-8z" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                        <circle cx="12" cy="12" r="3" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                      </svg>
                    </button>
                    <button className="btn-action" title="Editar" onClick={() => handleEditarServico(servico)}>
                      <svg width="16" height="16" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                        <path d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                        <path d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                      </svg>
                    </button>
                    <button className="btn-action" title="Remover" onClick={() => handleRemoverServico(servico)}>
                      <svg width="16" height="16" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
                        <polyline points="3,6 5,6 21,6" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                        <path d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                        <line x1="10" y1="11" x2="10" y2="17" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                        <line x1="14" y1="11" x2="14" y2="17" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
                      </svg>
                    </button>
                  </td>
                </tr>
              ))}
              {servicosFiltrados.length === 0 && (
                <tr>
                  <td colSpan="6" style={{ textAlign: 'center', padding: '20px' }}>
                    {servicos.length === 0 ? 'Nenhum serviço encontrado. Clique em "+ Novo Serviço" para adicionar.' : 'Nenhum serviço encontrado com os filtros selecionados.'}
                  </td>
                </tr>
              )}
            </tbody>
          </table>
        </div>
      )}

      {/* Modal de Edição de Serviço */}
      {showEditServico && (
        <>
          <div className="edit-overlay" />
          <div className="edit-servico-modal">
            <div className="edit-header">
              <h3>Editar Serviço</h3>
              <button className="btn-fechar-edit" onClick={handleCancelarEdicao}>
                ✕
              </button>
            </div>
            
            <form onSubmit={handleSalvarEdicao}>
              <div className="form-grid">
                <div className="form-group">
                  <label>Cliente</label>
                  <select
                    name="cliente"
                    value={editFormData.cliente || ''}
                    onChange={handleEditarChange}
                  >
                    <option value="">Selecione um cliente</option>
                    {clientes.map(cliente => (
                      <option key={cliente._id} value={cliente.razaoSocial || cliente.nome}>
                        {cliente.razaoSocial || cliente.nome}
                      </option>
                    ))}
                  </select>
                </div>
                
                <div className="form-group">
                  <label>Tipo de Serviço *</label>
                  <select
                    name="tipoServico"
                    value={editFormData.tipoServico || ''}
                    onChange={handleEditarChange}
                    
                  >
                    <option value="">Selecione o tipo</option>
                    <option value="Dedetização">Dedetização</option>
                    <option value="Descupinização">Descupinização</option>
                    <option value="Limpeza de Caixa D'água">Limpeza de Caixa D'água</option>
                    <option value="Desinsetização">Desinsetização</option>
                    <option value="Controle de Pragas">Controle de Pragas</option>
                  </select>
                </div>
                
                <div className="form-group">
                  <label>Status *</label>
                  <select
                    name="status"
                    value={editFormData.status || ''}
                    onChange={handleEditarChange}
                  >
                    <option value="">Selecione o status</option>
                    <option value="Executado">Executado</option>
                    <option value="Agendado">Agendado</option>
                    <option value="Vendido">Vendido</option>
                    <option value="Nao Agendado">Não Agendado</option>
                    <option value="Em execucao">Em execução</option>
                    <option value="Vencido">Vencido</option>
                    <option value="Cancelado">Cancelado</option>
                  </select>
                </div>
                
                <div className="form-group">
                  <label>Nº Orçamento</label>
                  <input
                    type="text"
                    name="numeroOrcamento"
                    value={editFormData.numeroOrcamento || ''}
                    onChange={handleEditarChange}
                    placeholder="Ex: ORC-001"
                  />
                </div>
              </div>
              
              <div className="form-grid">
                <div className="form-group">
                  <label>Valor do Serviço (R$) *</label>
                  <input
                    type="number"
                    name="valor"
                    value={editFormData.valor || ''}
                    onChange={handleEditarChange}
                    step="0.01"
                    min="0"
                    placeholder="0,00"
                  />
                </div>
                
                <div className="form-group">
                  <label>Periodicidade</label>
                  <select
                    name="periodicidade"
                    value={editFormData.periodicidade || ''}
                    onChange={handleEditarChange}
                  >
                    <option value="">Selecione</option>
                    <option value="Mensal">Mensal</option>
                    <option value="Bimestral">Bimestral</option>
                    <option value="Trimestral">Trimestral</option>
                    <option value="Semestral">Semestral</option>
                    <option value="Anual">Anual</option>
                    <option value="Único">Único</option>
                  </select>
                </div>
              </div>
              
              <div className="form-group">
                <label>Datas de Execução</label>
                {editFormData.datasExecucao && editFormData.datasExecucao.length > 0 ? (
                  <div className="datas-execucao-edit">
                    {editFormData.datasExecucao.map((data, index) => {
                      // Simplificar formatação de dados
                      let dataFormatada = '';
                      
                      try {
                        // Obter valor da data de forma segura
                        const dataValor = data?.data || data || '';
                        
                        if (!dataValor) {
                          dataFormatada = '';
                        } else if (typeof dataValor === 'string') {
                          if (dataValor.includes('T')) {
                            // Formato ISO
                            const dateObj = new Date(dataValor);
                            dataFormatada = dateObj.toLocaleDateString('pt-BR');
                          } else if (dataValor.includes('-')) {
                            // Formato yyyy-mm-dd
                            const [ano, mes, dia] = dataValor.split('-');
                            dataFormatada = `${dia}/${mes}/${ano}`;
                          } else if (dataValor.includes('/')) {
                            // Já está em formato brasileiro
                            dataFormatada = dataValor;
                          } else {
                            dataFormatada = '';
                          }
                        } else {
                          dataFormatada = '';
                        }
                        
                        // Verificar se resultado é válido
                        if (dataFormatada === 'Invalid Date' || dataFormatada === 'NaN/NaN/NaN' || !dataFormatada) {
                          dataFormatada = '';
                        }
                        
                      } catch (error) {
                        console.warn('Erro ao formatar data:', error);
                        dataFormatada = '';
                      }
                      
                      return (
                        <div key={index} className="data-execucao-item-edit">
                          <input
                            type="text"
                            placeholder="dd/mm/aaaa"
                            maxLength="10"
                            value={data?.data || dataFormatada || ''}
                            onChange={(e) => {
                              let value = e.target.value;
                              
                              // Permitir edição mesmo quando apagando
                              const numerosApenas = value.replace(/\D/g, '');
                              let formatado = '';
                              
                              if (numerosApenas.length > 0) {
                                formatado = numerosApenas;
                                
                                // Adicionar barras automaticamente
                                if (numerosApenas.length >= 2) {
                                  formatado = numerosApenas.slice(0, 2) + '/' + numerosApenas.slice(2);
                                }
                                if (numerosApenas.length >= 4) {
                                  formatado = formatado.slice(0, 5) + '/' + formatado.slice(5);
                                }
                                
                                // Limitar a 10 caracteres (dd/mm/yyyy)
                                if (formatado.length > 10) {
                                  formatado = formatado.slice(0, 10);
                                }
                              }
                              
                              console.log('Editando data índice', index, 'valor digitado:', value, 'formatado:', formatado);
                              
                              const novasDatas = [...editFormData.datasExecucao];
                              
                              // Garantir que o item exista e tenha estrutura correta
                              if (!novasDatas[index]) {
                                novasDatas[index] = {};
                              }
                              
                              // Atualizar sempre como objeto com propriedade 'data'
                              novasDatas[index] = {
                                ...novasDatas[index],
                                data: formatado, // Permite string vazia quando apagado
                                status: novasDatas[index]?.status || 'Programado',
                                observacao: novasDatas[index]?.observacao || ''
                              };
                              
                              console.log('Novas datas após edição:', novasDatas);
                              
                              setEditFormData(prev => ({
                                ...prev,
                                datasExecucao: novasDatas
                              }));
                            }}
                          />
                          <select 
                            className="status-data" 
                            value={data?.status || 'Programado'} 
                            onChange={(e) => {
                              const novasDatas = [...editFormData.datasExecucao];
                              if (!novasDatas[index]) {
                                novasDatas[index] = {};
                              }
                              novasDatas[index] = {
                                ...novasDatas[index],
                                data: novasDatas[index]?.data || dataFormatada || '',
                                status: e.target.value,
                                observacao: novasDatas[index]?.observacao || ''
                              };
                              setEditFormData(prev => ({
                                ...prev,
                                datasExecucao: novasDatas
                              }));
                            }}
                          >
                            <option value="Programado">Programado</option>
                            <option value="Executado">Executado</option>
                            <option value="Cancelado">Cancelado</option>
                          </select>
                          <input
                            type="text"
                            value={data?.observacao || ''}
                            onChange={(e) => {
                              const novasDatas = [...editFormData.datasExecucao];
                              if (!novasDatas[index]) {
                                novasDatas[index] = {};
                              }
                              novasDatas[index] = {
                                ...novasDatas[index],
                                data: novasDatas[index]?.data || dataFormatada || '',
                                status: novasDatas[index]?.status || 'Programado',
                                observacao: e.target.value
                              };
                              setEditFormData(prev => ({
                                ...prev,
                                datasExecucao: novasDatas
                              }));
                            }}
                            placeholder="Observação"
                          />
                          <button
                            type="button"
                            onClick={() => {
                              const novasDatas = editFormData.datasExecucao.filter((_, i) => i !== index);
                              setEditFormData(prev => ({
                                ...prev,
                                datasExecucao: novasDatas
                              }));
                            }}
                            className="btn-remover-data"
                          >
                            ✕
                          </button>
                        </div>
                      );
                    })}
                    <button
                      type="button"
                      onClick={() => {
                        setEditFormData(prev => ({
                          ...prev,
                          datasExecucao: [...(prev.datasExecucao || []), { data: '' }]
                        }));
                      }}
                      className="btn-adicionar-data"
                    >
                      + Adicionar Data
                    </button>
                  </div>
                ) : (
                  <div>
                    <input
                      type="text"
                      placeholder="dd/mm/aaaa"
                      maxLength="10"
                      value=""
                      onChange={(e) => {
                        let value = e.target.value;
                        
                        // Permitir apenas números e formatação automática
                        const numerosApenas = value.replace(/\D/g, '');
                        let formatado = '';
                        
                        if (numerosApenas.length > 0) {
                          formatado = numerosApenas;
                          
                          // Adicionar barras automaticamente
                          if (numerosApenas.length >= 2) {
                            formatado = numerosApenas.slice(0, 2) + '/' + numerosApenas.slice(2);
                          }
                          if (numerosApenas.length >= 4) {
                            formatado = formatado.slice(0, 5) + '/' + formatado.slice(5);
                          }
                          
                          // Limitar a 10 caracteres (dd/mm/yyyy)
                          if (formatado.length > 10) {
                            formatado = formatado.slice(0, 10);
                          }
                        }
                        
                        setEditFormData(prev => ({
                          ...prev,
                          datasExecucao: [{ data: formatado, status: 'Programado', observacao: '' }]
                        }));
                      }}
                    />
                  </div>
                )}
              </div>
              
              <div className="form-grid">
                <div className="form-group">
                  <label>Valor do Serviço (R$)</label>
                  <input
                    type="number"
                    name="valor"
                    value={editFormData.valor || ''}
                    onChange={handleEditarChange}
                    step="0.01"
                    min="0"
                    placeholder="0,00"
                  />
                </div>
                
                <div className="form-group">
                  <label>Próxima Execução</label>
                  <input
                    type="text"
                    name="proximaExecucao"
                    placeholder="dd/mm/aaaa"
                    maxLength="10"
                    value={editFormData.proximaExecucao || ''}
                    onChange={(e) => {
                      let value = e.target.value;
                      
                      // Aplicar máscara dd/mm/yyyy
                      value = value.replace(/\D/g, ''); // Remove não números
                      if (value.length >= 2) {
                        value = value.slice(0, 2) + '/' + value.slice(2);
                      }
                      if (value.length >= 5) {
                        value = value.slice(0, 5) + '/' + value.slice(5);
                      }
                      if (value.length > 10) {
                        value = value.slice(0, 10);
                      }
                      
                      setEditFormData(prev => ({
                        ...prev,
                        proximaExecucao: value
                      }));
                    }}
                  />
                </div>
              </div>
              
              <div className="form-grid">
                <div className="form-group">
                  <label>Horário</label>
                  <input
                    type="text"
                    name="horario"
                    value={editFormData.horario || ''}
                    onChange={handleEditarChange}
                    placeholder="Ex: 08:00"
                  />
                </div>
                
                <div className="form-group">
                  <label>Local / Endereço</label>
                  <input
                    type="text"
                    name="localEndereco"
                    value={editFormData.localEndereco || ''}
                    onChange={handleEditarChange}
                    placeholder="Endereço do serviço"
                  />
                </div>
              </div>
              
              <div className="form-grid">
                <div className="form-group">
                  <label>Cidade</label>
                  <input
                    type="text"
                    name="cidade"
                    value={editFormData.cidade || ''}
                    onChange={handleEditarChange}
                    placeholder="Cidade do serviço"
                  />
                </div>
                
                <div className="form-group">
                  <label>Supervisor Responsável</label>
                  <input
                    type="text"
                    name="supervisorResponsavel"
                    value={editFormData.supervisorResponsavel || ''}
                    onChange={handleEditarChange}
                    placeholder="Nome do supervisor"
                  />
                </div>
              </div>
              
              <div className="form-grid">
                <div className="form-group">
                  <label>Equipe Membros</label>
                  <input
                    type="text"
                    name="equipeMembros"
                    value={editFormData.equipeMembros || ''}
                    onChange={handleEditarChange}
                    placeholder="Nomes separados por vírgula"
                  />
                </div>
                
                <div className="form-group">
                  <label>Veículo / Placa</label>
                  <input
                    type="text"
                    name="veiculoPlaca"
                    value={editFormData.veiculoPlaca || ''}
                    onChange={handleEditarChange}
                    placeholder="Placa do veículo"
                  />
                </div>
              </div>
              
              <div className="form-grid">
                <div className="form-group">
                  <label>Estruturas Realizadas</label>
                  <input
                    type="text"
                    name="estruturasRealizadas"
                    value={editFormData.estruturasRealizadas || ''}
                    onChange={handleEditarChange}
                    placeholder="Descreva as estruturas tratadas"
                  />
                </div>
                
                <div className="form-group">
                  <label>Acompanhante / Assinatura</label>
                  <input
                    type="text"
                    name="acompanhanteAssinatura"
                    value={editFormData.acompanhanteAssinatura || ''}
                    onChange={handleEditarChange}
                    placeholder="Nome do acompanhante"
                  />
                </div>
              </div>
              
              <div className="form-group">
                <label>Descrição do Serviço</label>
                <textarea
                  name="descricao"
                  value={editFormData.descricao || ''}
                  onChange={handleEditarChange}
                  rows="3"
                  placeholder="Descreva detalhes do serviço..."
                />
              </div>
              
              <div className="form-actions">
                <button type="button" className="btn-secondary" onClick={handleCancelarEdicao}>
                  Cancelar
                </button>
                <button type="submit" className="btn-primary">
                  Salvar Alterações
                </button>
              </div>
            </form>
          </div>
        </>
      )}
      
      {/* Janela Flutuante de Relatório */}
      <RelatoriosServicos 
        servico={servicoSelecionado}
        isVisible={showRelatorio}
        onClose={() => setShowRelatorio(false)}
      />
    </div>
  );
};
export default Servicos;
